dirty_cat/datasets/fetching.py

```diff
@@ -197,33 +196,32 @@ def fetch_world_bank_data(dataset_id: str, indicator: str,
               saved as a CSV file.
 
     """
-
-    # Make path absolute
-    if not os.path.exists(data_directory):
-        os.makedirs(data_directory)
-    data_directory = data_directory.resolve() / 'world_bank.zip'
-    zip_path = Path(str(data_directory) + '_folder')
+    # Create directory if non-existant:
+    data_directory.mkdir(exist_ok=True, parents=True)
+    data_directory = data_directory.resolve() / "world_bank"
+    zip_path = Path(str(data_directory) + "_folder")
     # Download the file :
-    url = f'https://api.worldbank.org/v2/en/indicator/{dataset_id}?downloadformat=csv'
+    url = f"https://api.worldbank.org/v2/en/indicator/{dataset_id}?downloadformat=csv"
     urllib.request.urlretrieve(url, data_directory)
-    # Extract csv file :
-    with ZipFile(data_directory, 'r') as f:
-        names = f.namelist()
-        for n in names:
-            if 'Metadata' not in n:
-                f.extract(
-                    n, path=zip_path)
-                true_file = n
-    f.close()
+    try:
+        # Extract csv file :
+        with ZipFile(data_directory, "r") as f:
+            names = f.namelist()
+            for n in names:
+                if "Metadata" not in n:
+                    f.extract(n, path=zip_path)
+                    true_file = n
+        f.close()
+    except:
+        raise FileNotFoundError(f"Couldn't find file {data_directory!s}, please check dataset_id.")
```

Comment:
Errors should guide the user as much as possible. For example, if the function was used in a complex pipeline with hundreds of steps, it would be hard to know what `dataset_id` might be. 
Printing the value directly often helps finding the source of the issue.
```suggestion
        raise FileNotFoundError(f"Couldn't find directory {data_directory!s}, dataset id {dataset_id!r} seems invalid.")
```
Also, the usage of `directory` and `file` in this whole code block is ambiguous. Despite working kind of like a directory, a zip is a file.